src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventHandler.cs

```diff
@@ -79,7 +85,17 @@ public void ProcessRequests()
                     }
 
                     var testCaseStartEventArgs = _dataSerializer.DeserializePayload<TestCaseStartEventArgs>(message);
-                    _dataCollectionManager.TestCaseStarted(testCaseStartEventArgs);
+
+                    try
+                    {
+                        _dataCollectionManager.TestCaseStarted(testCaseStartEventArgs);
+                    }
+                    catch (Exception ex)
+                    {
+                        _messageSink.SendMessage(new DataCollectionMessageEventArgs(TestMessageLevel.Error, $"Error occurred during TestCaseStarted event handling: {ex}"));
```

Comment:
Probably doesn't matter much but you could extract the `ex.ToString()` in a variable instead of forcing the call twice.

Reply:
it should never happen(only in case of bug), we can keep as-is for now one less line.